packages/sanity/src/core/form/studio/assetSource/DefaultSource.tsx

```diff
@@ -14,8 +14,51 @@ const PER_PAGE = 200
 const ASSET_TYPE_IMAGE = 'sanity.imageAsset'
 const ASSET_TYPE_FILE = 'sanity.fileAsset'
 
-const buildQuery = (start = 0, end = PER_PAGE, assetType = ASSET_TYPE_IMAGE) => `
-  *[_type == "${assetType}"] | order(_updatedAt desc) [${start}...${end}] {
+const buildFilterQuery = (acceptParam: string) => {
+  const WILDCARD_ACCEPT = ['image/*', 'audio/*', 'video/*']
+  const acceptItems = acceptParam.split(',').map((accept) => accept.trim())
+
+  const typesForFilter: {[key: string]: string} = acceptItems.reduce(
+    (acceptTypes: {[key: string]: string}, acceptR: string) => {
+      // builds the wildcard part of the groq query
+      if (WILDCARD_ACCEPT.includes(acceptR)) {
+        const wild = `mimeType match '${acceptR}' || ${acceptTypes.wildcards}`
+        return {...acceptTypes, ...{wildcards: wild}}
+      }
+
+      // builds the extension part of the groq query (and removes the .)
+      if (acceptR.indexOf('.') === 0) {
+        const exte = `'${acceptR.replace('.', '')}', ${acceptTypes.mimes}`
+        return {...acceptTypes, ...{extensions: exte}}
+      }
+
+      // all that remains is then the mime types, so we build that part
+      const mime = `'${acceptR}', ${acceptTypes.mimes}`
+      return {...acceptTypes, ...{mimes: mime}}
+    },
+    {mimes: '', extensions: '', wildcards: ''}
+  )
+
+  // when no accept filter is set, we don't need to add the filter condition
+  return `&&
+  (
+    ${typesForFilter.wildcards} 
+    extension in [${typesForFilter.extensions}] ||
```

Comment:
if there's no extension this will result in `extension in ['']` - will that work as expected?

Reply:
Yes, it will!
(so in the use case `audio/*, video/*, image/jpeg, image/png`, it'll show all audios, videos, jpegs and pngs).
I understand it makes the query a bit noisy, I considered adding a specific condition to tackle these extensions but thought it might just add noise considering it works as expected 🤔 What do you think?